vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/components/ExperimentalVariableCardCollection.java

```diff
@@ -0,0 +1,129 @@
+package life.qbic.datamanager.views.projects.project.experiments.experiment.components;
+
+import com.vaadin.flow.component.ClickEvent;
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Span;
+import java.io.Serial;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import life.qbic.datamanager.views.general.AddEvent;
+import life.qbic.datamanager.views.general.EditEvent;
+import life.qbic.datamanager.views.projects.project.experiments.experiment.ExperimentalVariableCard;
+import life.qbic.projectmanagement.domain.project.experiment.ExperimentalVariable;
+
+/**
+ * <b>Experimental Variable Collection</b>
+ * <p>
+ * Container of one or more {@link ExperimentalVariableCard}
+ *
+ * @since 1.0.0
+ */
+
+public class ExperimentalVariableCardCollection extends Div {
+
+  @Serial
+  private static final long serialVersionUID = -9123769128332512326L;
+
+  private final Div content = new Div();
+  private final List<ComponentEventListener<AddEvent<ExperimentalVariableCardCollection>>> addListeners = new ArrayList<>();
+  private final List<ComponentEventListener<EditEvent<ExperimentalVariableCardCollection>>> editListeners = new ArrayList<>();
```

Comment:
You do not need those lists. Have a look at [Component#addListener](https://vaadin.com/api/platform/current/com/vaadin/flow/component/Component.html#addListener(java.lang.Class,com.vaadin.flow.component.ComponentEventListener)) and [Component#fireEvent](https://vaadin.com/api/platform/current/com/vaadin/flow/component/Component.html#fireEvent(com.vaadin.flow.component.ComponentEvent))